service/static/js/rest_api.js

```diff
@@ -393,5 +435,176 @@ $(function () {
+    });
+
+    // ****************************************
+    // Update a Address
+    // ****************************************
+    $("#addr-update-btn").click(function () {
+        var customer_id = $("#cust_id").val();
+        var address_id = $("#addr_id").val();
+        
+        var street_address = $("#addr_street_address").val();
+        var city = $("#addr_city").val();
+        var state = $("#addr_state").val();
+        var country = $("#addr_country").val();
+        var zipcode = $("#addr_zip").val();
+
+        var data = {
+            "street_address": street_address,
+            "city":city,
+            "state": state,
+            "zipcode": zipcode,
+            "country": country
+        };
+
+        var ajax = $.ajax({
+                type: "PUT",
+                url: "/customers/" + customer_id+"/addresses/" +address_id,
+                contentType: "application/json",
+                data: JSON.stringify(data)
+            })
+
+        ajax.done(function(res){
+            res.id = customer_id
+            update_address_form_data(res)
+            flash_message("Success")
+        });
+
+        ajax.fail(function(res){
+            flash_message(res.responseJSON.message)
+        });
+
+    });
+
+    // ****************************************
+    // Retrieve an Address
+    // ****************************************
+    $("#addr-search-btn").click(function () {
+
+        var customer_id = $("#cust_id").val();
+        var address_id = $("#addr_id").val();
+
+        var ajax = $.ajax({
+            type: "GET",
+            url: "/customers/" + customer_id+"/addresses/" +address_id,
+            contentType: "application/json",
+        })
+
+        ajax.done(function(res){
+            clear_form_data()
+            add_single_address(res)
+            update_address_form_data(res)
+            flash_message("Success")
+        });
+
+        ajax.fail(function(res){
+            clear_form_data()
+            flash_message(res.responseJSON.message)
+        });
+
+    });
+
+    /*
+    //Haven't implemented yet
+    // ****************************************
+    // Delete an Address
+    // ****************************************
+    $("#addr-delete-btn").click(function () {
+        
+        var customer_id = $("#cust_id").val();
+        var address_id = $("#addr_id").val();
+
+        var ajax = $.ajax({
+            type: "DELETE",
+            url: "/customers/" + customer_id+"/addresses/" +address_id,
+            contentType: "application/json"
+        })
+
+        ajax.done(function(res){
+            clear_form_data()
+            flash_message("Address has been Deleted!")
+        });
+
+        ajax.fail(function(res){
+            flash_message("Server error!")
+        });
+    });
+
+*/
+    // ****************************************
+    // //RETRIEVE A CUSTOMER'S ADDRESSES
+    // ****************************************
+    $("#addr-query-btn").click(function () {
```

Comment:
our api supports query parameters for addresses, and so I think it would be good to add support for that to this function

Reply:
i also think that within the form, we should make sure that the customer id is not erased, and can also have the first address in the customer's list of address show up in the form (that way it's similar to how "Search for Customer" results show

Reply:
Yes that's where I misunderstood the function, I will fix them.